gui/src/controls/complexLine.ts

```diff
@@ -0,0 +1,294 @@
+/// <reference path="../../../dist/preview release/babylon.d.ts"/>
+
+module BABYLON.GUI {
+
+    export class ComplexLine extends Control {
+
+        private _lineWidth: number = 1;
+        private _dash: number[];
+        private _trackers: Nullable<Tracker>[];
+
+        private _minX: Nullable<number>;
+        private _minY: Nullable<number>;
+        private _maxX: Nullable<number>;
+        private _maxY: Nullable<number>;
+
+        constructor(public name?: string) {
+            super(name);
+
+            this.isHitTestVisible = false;
+            this._horizontalAlignment = Control.HORIZONTAL_ALIGNMENT_LEFT;
+            this._verticalAlignment = Control.VERTICAL_ALIGNMENT_TOP;
+
+            this._dash = [];
+            this._trackers = [];
+        }
+
+        public get dash(): Array<number> {
+            return this._dash;
+        }
+
+        public set dash(value: Array<number>) {
+            if (this._dash === value) {
+                return;
+            }
+
+            this._dash = value;
+            this._markAsDirty();
+        }
+
+        getX(index: number): string | number  {
+            return this.getTracker(index).x.toString(this._host);
+        }
+
+        setX(index: number, value: string | number): void {
+            var point: Tracker = this.getTracker(index);
+
+            if (point.x.toString(this._host) === value) {
+                return;
+            }
+
+            if (point.x.fromString(value)) {
+                this._markAsDirty();
+            }
+        }
+
+        getY(index: number): string | number  {
+            return this.getTracker(index).y.toString(this._host);
+        }
+
+        setY(index: number, value: string | number): void {
+            var tracker: Tracker = this.getTracker(index);
+
+            if (tracker.y.toString(this._host) === value) {
+                return;
+            }
+
+            if (tracker.y.fromString(value)) {
+                this._markAsDirty();
+            }
+        }
+
+        getControl(index: number): Nullable<Control> {
+            return this.getTracker(index).control;
+        }
+
+        setControl(index: number, value: Nullable<Control> = null) {
+            var tracker: Tracker = this.getTracker(index);
+
+            if (tracker.control === value) {
+                return;
+            }
+
+            if (tracker.control) {
+                tracker.control.onDirtyObservable.removeCallback(this.onTrackerUpdate);
+            }
+
+            tracker.control = value;
+
+            if (tracker.control) {
+                tracker.control.onDirtyObservable.add(this.onTrackerUpdate);

```

Comment:
this may not work as "this" will not be the same in the observable. You may have to use () =>

Reply:
I did in the definition:
private onTrackerUpdate = (): void => {
            this._markAsDirty();
        }

That way the callback reference is the same so I can remove it easily later as well

Reply:
If private it has to start with_ then

Reply:
No problem